icicle/backend/cpu/include/cpu_sumcheck.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <vector>
+#include <functional>
+#include "icicle/program/symbol.h"
+#include "icicle/program/program.h"
+#include "icicle/backend/sumcheck_backend.h"
+
+namespace icicle {
+
+  template <typename F>
+  class CpuSumcheckBackend : public SumcheckBackend<F>
+  {
+  public:
+    CpuSumcheckBackend(const F& claimed_sum, SumcheckTranscriptConfig<F>&& transcript_config)
+        : SumcheckBackend<F>(claimed_sum, std::move(transcript_config))
+    {
+    }
+
+    eIcicleError get_proof(
+      const std::vector<std::vector<F>*>& input_polynomials,
```

Comment:
(1) maybe rename to mle?
(2) pointer to vector is strange. Why do you need that?

Reply:
recommend: Input MLE's